src/server/service/diagnostics/log.rs

```diff
@@ -155,44 +157,57 @@ impl Iterator for LogIterator {
                         continue;
                     }
                 };
-                if input.len() < TIMESTAMP_LENGTH {
+                if self.pre_log.time < self.begin_time && input.len() < TIMESTAMP_LENGTH {
                     continue;
                 }
+                let mut item = LogMessage::default();
                 match parse(&input) {
                     Ok((content, (time, level))) => {
-                        // The remain content timestamp more the end time or this file contains inrecognation formation
-                        if time == INVALID_TIMESTAMP || time > self.end_time {
-                            break;
-                        }
-                        if time < self.begin_time {
-                            continue;
-                        }
-                        if self.level_flag != 0 && self.level_flag & (1 << (level as usize)) == 0 {
-                            continue;
-                        }
-                        if !self.patterns.is_empty() {
-                            let mut not_match = false;
-                            for pattern in self.patterns.iter() {
-                                if !pattern.is_match(content) {
-                                    not_match = true;
-                                    break;
-                                }
-                            }
-                            if not_match {
-                                continue;
-                            }
-                        }
-                        let mut item = LogMessage::default();
+                        self.pre_log.set_time(time);
+                        self.pre_log.set_level(level);
+
                         item.set_time(time);
                         item.set_level(level);
                         item.set_message(content.to_owned());
-                        return Some(item);
                     }
                     Err(err) => {
                         warn!("parse line failed: {:?}", err);
+                        if self.pre_log.time < self.begin_time {
+                            continue;
+                        }
+                        // treat the invalid log with the pre valid log time and level but its own whole line content
+                        item.set_time(self.pre_log.time);
+                        item.set_level(self.pre_log.level as LogLevel);
```

Comment:
Change here and L192 to `logitem.get_level()` will fix lints. 
The reason for failing CI is that it will enable two codec features separately and run tests, one is `protobuf-codec`, the other is `prost-codec`. These two codec platforms are slightly different on APIs.

Reply:
Run `PROST=1 make clippy` to check with the `prost-codec` feature.

Reply:
Nice, let me have a try, thanks!